src/pair.py

```diff
@@ -143,3 +143,96 @@ def __init__(self, nlist, default_r_cut=None, default_r_on=0, mode='none'):
             ),
         )
         self._add_typeparam(params)
+
+
+class TwoPatchMorse(pair.aniso.AnisotropicPair):
+    r"""TwoPatchMorse potential.
+
+    Args:
+        nlist (hoomd.md.nlist.NeighborList): Neighbor list.
+        default_r_cut (float): Default cutoff radius :math:`[\mathrm{length}]`.
+        mode (str): Energy shifting/smoothing mode.
+
+    :py:class:`TwoPatchMorse` is a Morse potential which is modulated by an
+    orientation-dependent function.
+
+    .. math::
+
+        V_{\rm M2P}(\vec{r}_{ij}, \hat{n}_i, \hat{n}_j) = V_{\rm M}(|\vec{r}_{ij}|)
+        \Omega(\hat{r}_{ij} \cdot \hat{n}_i) \Omega(\hat{r}_{ij} \cdot \hat{n}_j)
+
+    where :math:`V_{\rm M}` is the potential that depends on distance
+
+    .. math::
+
+        V_{\rm M}(r) = M_d \left( \left[ 1 - \exp\left(
+              -\frac{r-r_{\rm eq}}{M_r}\right) \right]^2 - 1 \right)
+
+    and :math:`\Omega(\gamma)` depends on the orientations
+
+    .. math::
+        \Omega(\gamma) = \frac{1}{1+\exp[-\omega (\gamma^2 - \alpha)]}
+
+    The potential can be smoothed to zero force (making it purely attractive)
+    when :math:`r < r_{\rm eq}` by making :math:`V_{\rm M}(r < r_{\rm eq}) = -M_d`
+    when the option  *repulsion* is ``False``.
+
+    Here, :math:`vec{r}_{ij}` is the displacement vector between particles
+    :math:`i` and :math:`j`, :math:`|\vec{r}_{ij}|` is the magnitude of
+    that displacement, and :math:`\hat{n}` is the normalized
+    orientation vector of the particle. The parameters :math:`M_d`,
+    :math:`M_r`, and :math:`r_{\rm eq}` control the depth, width, and
+    position of the potential well. The parameters :math:`\alpha` and
+    :math:`\omega` control the width and steepness of the orientation dependence.
+
+    Example::
+
+        nl = hoomd.md.nlist.cell()
+        m2p = azplugins.pair.TwoPatchMorse(r_cut=1.6, nlist=nl)
+        m2p.pair_coeff.set('A', 'A', Mdeps=1.8347, Mr=0.0302, req=1.0043,
+        omega=20, alpha=0.50, repulsion=True)
+
+    .. py:attribute:: params
+
+        The Two Patch Morse potential parameters. The dictionary has the following
+        keys:
+
+        * ``Mdeps`` (`float`, **required**) - controls the depth of the
+         potential well
+          :math:`M_d` :math:`[\mathrm{energy}]`
+        * ``Mr`` (`float`, **required**) - controls the width of the
+         potential well :math:`M_r` :math:`[\mathrm{length}]`
+        * ``req`` (`float`, **required**) - controls the position of
+        the potential well :math:`r_eq` :math:`[\mathrm{length}]`
```

Comment:
I think we should rename these parameters based on how the equations are structured (not sure why they were the way they were before). Please update here, and then also in the header file.
```suggestion
        * ``M_d`` (`float`, **required**) - controls the depth of the
         potential well
          :math:`M_d` :math:`[\mathrm{energy}]`
        * ``M_r`` (`float`, **required**) - controls the width of the
         potential well :math:`M_r` :math:`[\mathrm{length}]`
        * ``r_eq`` (`float`, **required**) - controls the position of
        the potential well :math:`r_eq` :math:`[\mathrm{length}]`
```

Reply:
How would you like Mrinv to be renamed? M_rinv?